### DISCUSSION TITLE
Referencing a heading with it's "value" as content.

### DISCUSSION THREAD

**Discussion by CasparNuel** [typst/typst#2656] [Q&A] [answered]
I am trying to achieve a reference to a heading (or any element really) that uses it's "content" or "title" (not sure how to call it) as part of the reference. This is what I tried so far:

I have numbered headings in my document. This is an mwe:

```typ
#set heading(numbering: "1.1  ")

Read more about the awesome and epic stuff in @awesome , @epic

= This is an awesome chapter! <awesome>

= This is an epic chapter! <epic>
```

This results in:

![image](https://github.com/typst/typst/assets/73889285/286a0f3a-09e3-4891-aba8-ada3e79683eb)

I am actually trying to find a way to get it to look like this (currently just hard coded to demonstrate my goal):

![image](https://github.com/typst/typst/assets/73889285/8a916a58-6809-45bd-a9fe-22250805d5d3)

Ideally this is seperate from the `supplement` field so one could rename "sections" to "chapters" without needing to change the heading value itself (whatever comes after the `=`).

How can I achieve this?

**Comment by xkevio** [comment, marked as answer]
```typ
#show ref: r => {
  if r.element != none and r.element.func() == heading {
    link(r.element.location())[#r #emph(r.element.body)]
  } else {
    r
  }
}
```

**Reply by CasparNuel** [reply]
Thanks a bunch! I have been getting better at the scripting language but sometimes I don't see the correct path right away. :smile:
